pyuvdata/uvdata/uvdata.py

```diff
@@ -2360,6 +2380,167 @@ def _check_freq_spacing(self, raise_errors=True):
             raise_errors=raise_errors,
         )
 
+    def remove_flex_pol(self):
+        """
+        Convert a flex-pol UVData object into one with a standard polarization axis.
+
+        This will convert a flexible-polarization dataset into one with standard
+        polarization handling, which is required for some operations or writing in
+        certain filetypes. Note that depending on how it is used, this can inflate
+        the size of data-like parameters by up to a factor of Nspws (the true value
+        depends on the number of unique entiries in `flex_spw_polarization_array`).
+        """
+        if self.flex_spw_polarization_array is None:
+            # There isn't anything to do, so just move along
+            return
+
+        self.polarization_array = np.unique(self.flex_spw_polarization_array)
+        self.Npols = len(self.polarization_array)
+
+        # If we have metadata only, or there was only one pol we were working with,
+        # then we do not need to do anything further aside from removing the array
+        # associated with flex_spw_polarization_array
+        if self.metadata_only or (self.Npols == 1):
+            self.flex_spw_polarization_array = None
+            return
+
+        # Otherwise, move through all of the data params
+        for name, param in zip(self._data_params, self.data_like_parameters):
+            # We need to construct arrays with the appropriate shape
+            new_shape = [self.Nblts, 1, self.Nfreqs, self.Npols]
+
+            # Pop the spw-axis if using future array shapes
+            if self.future_array_shapes:
+                del new_shape[1]
+
+            # Use fill here, since we want zeros if we are working with nsample_array
```

Comment:
I might be getting pedantic, but it says "fill", but actually uses "full"

Reply:
I blame the good folks at numpy. Fixed in 8f8a8af4.